devicetypes/smartthings/zwave-door-temp-sensor.src/zwave-door-temp-sensor.groovy

```diff
@@ -0,0 +1,299 @@
+    }
+
+    result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+    def event = createEvent(descriptionText: "${device.displayName} woke up", isStateChange: false)
+    def cmds = []
+
+    // If any of our initial request didn't make it, request the sensor data again
+    if (!state.MSR) {
+        cmds << zwave.manufacturerSpecificV2.manufacturerSpecificGet()
+    }
+
+    if (device.currentValue("contact") == null) {
+        cmds << zwave.sensorBinaryV2.sensorBinaryGet(sensorType: zwave.sensorBinaryV2.SENSOR_TYPE_DOOR_WINDOW)
+    }
+
+    if ((!state.lastbat) || ((now() - state.lastbat) > (53*60*60*1000))) {
+        cmds << zwave.batteryV1.batteryGet()
+    }
+
+    cmds << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1, scale: 1)
+    cmds << zwave.wakeUpV1.wakeUpNoMoreInformation()
+
+    [event, response(commands(cmds, 1000))]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+    log.debug "SensorMultilevelReport: $cmd"
+    def map = [:]
+    switch (cmd.sensorType) {
+        case 1:
+            map.name = "temperature"
+            def cmdScale = cmd.scale == 1 ? "F" : "C"
+            def realTemperature = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+            map.value = getAdjustedTemp(realTemperature)
+            map.unit = getTemperatureScale()
+            log.debug "Temperature Report: $map.value"
+            break
+        default:
+            map.descriptionText = cmd.toString()
+            break
+    }
+
+    [createEvent(map)]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+    def map = [ name: "battery", unit: "%" ]
+    if (cmd.batteryLevel == 0xFF) {
+        map.value = 1
+        map.descriptionText = "${device.displayName} has a low battery"
+        map.isStateChange = true
+    } else {
+        map.value = cmd.batteryLevel
+    }
+
+    state.lastbat = now()
+
+    [createEvent(map)]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+    def result = []
+    def msr = String.format("%04X-%04X-%04X", cmd.manufacturerId, cmd.productTypeId, cmd.productId)
+
+    log.debug "msr: $msr"
+    updateDataValue("MSR", msr)
+
+    result << createEvent(descriptionText: "$device.displayName MSR: $msr", isStateChange: false)
+
+    result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand(commandClassVersions)
+
+    if (encapsulatedCommand) {
+        state.sec = 1
+        zwaveEvent(encapsulatedCommand)
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+    createEvent(descriptionText: "$device.displayName: $cmd", displayed: false)
+}
+
+private command(physicalgraph.zwave.Command cmd) {
+    def zwInfo = zwaveInfo
+
+    if ((zwInfo?.zw == null && state.sec) ||
```

Comment:
I think state.sec can be removed here as well and solely rely on zwInfo.zw.contains("s")